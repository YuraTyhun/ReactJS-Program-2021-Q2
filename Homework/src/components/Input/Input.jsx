import './Input.scss';
import React from 'react';

const Input = () => (
    <input 
        type="text"
        className="search-input-field"
        placeholder="What do you want to watch?" />
);

export default Input;

