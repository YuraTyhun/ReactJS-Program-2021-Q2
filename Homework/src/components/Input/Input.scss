.search-input-field {
    width: 100%;
    padding: 10px 15px;
    margin-right: 10px;
    font-size: 20px;
    color: #ffffff;
    border: none;
    border-radius: 5px;
    background-color: #424242;
}