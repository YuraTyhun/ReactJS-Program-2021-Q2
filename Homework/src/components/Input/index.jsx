import React from 'react';
import PropTypes from 'prop-types';

import './Input.scss';

const Input = ({ onChange }) => (
  <input
    type="text"
    className="search-input-field"
    placeholder="What do you want to watch?"
    onChange={onChange}
  />
);

Input.propTypes = {
  onChange: PropTypes.func.isRequired
};

export default Input;
