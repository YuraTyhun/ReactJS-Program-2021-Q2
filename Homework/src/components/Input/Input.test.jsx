import React from 'react';
import renderer from 'react-test-renderer';

import Input from './index';

it('Input renders correctly', () => {
    const input = renderer
        .create(<Input value="test value" onChange={() => {}}/>)
        .toJSON();
    expect(input).toMatchSnapshot();
})