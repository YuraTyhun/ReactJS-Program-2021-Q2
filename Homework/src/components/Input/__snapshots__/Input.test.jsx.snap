// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly 1`] = `
<input
  className="search-input-field"
  onChange={[Function]}
  placeholder="What do you want to watch?"
  type="text"
  value="test value"
/>
`;
